scene/gui/line_edit.cpp

```diff
@@ -1408,9 +1413,11 @@ void LineEdit::copy_text() {
 }
 
 void LineEdit::cut_text() {
-	if (editable && selection.enabled && !pass) {
+	if (selection.enabled && !pass) {
 		DisplayServer::get_singleton()->clipboard_set(get_selected_text());
-		selection_delete();
+		if (editable) {
+			selection_delete();
+		}
```

Comment:
This is wrong. Cutting should always be a complete operation.

Reply:
I'll change it to use `copy_text` instead and revert the change of the function.

Reply:
No, the Cut shortcut just should do nothing.

Reply:
Should the behavior of `TextEdit` be changed to match this one? (In a seperate pull request)

Reply:
Yes.

Also we could add tests for these fixes.

Reply:
May I ask why it should be reverted, since it just got added in april? (https://github.com/godotengine/godot/pull/86978)

Reply:
tbh this can be discussed further.

I think cut not actually cutting text, but only copying it, is wrong. Cut is an operation that removes text **and** copies it to clipboard, so it should not be doing one of them. As I noted https://github.com/godotengine/godot/pull/99822#discussion_r1863946132, that's how readonly fields behave in browsers, which is a nice reference for how things should work.

That said, TextEdit even has text for this behavior, so maybe it's better to open an issue about it. Currently the Cut operation is disabled in menu, so it's weird that it would do something via shortcut.
![image](https://github.com/user-attachments/assets/d229738d-743b-475c-bde1-2ae8da134677)